### ISSUE TITLE
[BUG] Memory Leak with GRContext.SetResourceCacheLimit

### ISSUE THREAD

**Issue by imerzan** [mono/SkiaSharp#1947]
**Description**

I am using an SKGLControl in my Windows Forms app to draw sections of a Bitmap. It all works fine but I get really low FPS in my app when the bitmap sizes are big (5000x5000 or higher dimensions, sometimes 25MB+ size. I instantiate the bitmap(s) once for the life of the application, so this is not the source of the mem leak.

**Code**

I used the following code to help with the larger bitmap files, and I instantly went from 10 fps to 60+ fps. This really helps. Except it causes a mem leak.

```csharp
// This runs once during application startup
while (mapCanvas.GRContext is null) await Task.Delay(1); // wait for GRContext to be set
mapCanvas.GRContext.SetResourceCacheLimit(1500 * 1000 * 1000); // Gives a big FPS boost on bigger bitmaps
```
bitmap drawing code
```csharp
        private readonly SKPaint _mapPaint = new SKPaint() // Removing this = no change
        {
            IsAntialias = true,
            FilterQuality = SKFilterQuality.Medium
        };
~snip~

canvas.DrawBitmap(bitmap, bounds, dest, _mapPaint);
```
**Expected Behavior**

I expect that this wouldn't leak memory.

**Actual Behavior**

If I comment out this line of code: 
```csharp
mapCanvas.GRContext.SetResourceCacheLimit(1500 * 1000 * 1000); // Gives a big FPS boost on bigger bitmaps
```
there is no leak. If I uncomment this line of code, there IS a leak.

**Basic Information**

- Version with issue:  2.80.3
- Last known good version:  unknown
- IDE:  VS2022 Latest version
- Platform Target Frameworks: 
  - Windows Classic:  .NET6 x64
  
<details>
  <summary>Running memory profile shows the memory steadily increasing until it reaches ~95% memory used on the system (9GB+). Managed Memory shows no leaks, it appears to be in unmanaged memory. When i comment out the line of code above, I stay under ~1.5GB memory used.</summary>

**Comment by LeetPro360** [comment]
Same issue for me, high mem leak

**Comment by imerzan** [comment]
So just wanted to provide an update on this... Previously I used 
`SKCanvas.GRContext.PurgeResources()` and this seemed to prevent Memory from spiraling out of control.

Today (months later) I removed this line to see if memory was still leaking, and apparently it is not. Not sure what was causing my app to gobble up all of my memory months ago, and why PurgeResources() seemed to fix it, but now there is no problem. Hmm.

Will close this since it's no longer an issue.
